server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -126,7 +127,11 @@ public static long sizeOfRelocatingShards(
 
         // Where reserved space is unavailable (e.g. stats are out-of-sync) compute a conservative estimate for initialising shards
         for (ShardRouting routing : node.shardsWithState(ShardRoutingState.INITIALIZING)) {
-            if (routing.relocatingNodeId() == null) {
+            IndexMetadata indexMetadata = metadata != null ? metadata.index(routing.index()) : null;
```

Comment:
I think we can use `getIndexSafe` as in `getExpectedShardSize` and avoid the null checking? I think we should always have `metadata` here.